core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,233 @@
+    private int deaths = 0;
+    private int killstreak = 0;
+    private int killstreakMax = 0;
+    private int longestBowKill = 0;
+
+    public void onMurder() {
+      kills++;
+      killstreak++;
+      if (killstreak > killstreakMax) killstreakMax = killstreak;
+    }
+
+    public void onDeath() {
+      deaths++;
+      killstreak = 0;
+    }
+
+    public void setLongestBowKill(double distance) {
+      if (distance > longestBowKill) {
+        longestBowKill = Integer.parseInt(new DecimalFormat("#").format(Math.round(distance)));
+      }
+    }
+
+    public Component getBasicStatsMessage() {
+      String KD;
+      if (deaths == 0) {
+        KD = "0";
+      } else {
+        DecimalFormat df = new DecimalFormat("#.##");
+        KD = df.format(kills / deaths);
+      }
+      return new Component(
+          new PersonalizedTranslatable(
+                  "stats.basic",
+                  new PersonalizedText(Integer.toString(kills), ChatColor.GREEN),
+                  new PersonalizedText(Integer.toString(killstreak), ChatColor.GREEN),
+                  new PersonalizedText(Integer.toString(deaths), ChatColor.RED),
+                  new PersonalizedText(KD, ChatColor.GREEN))
+              .render());
+    }
+  }
+
+  public static Map<String, playerStats> allPlayerStats;
+
+  public StatsMatchModule() {
+    allPlayerStats = new HashMap<>();
+  }
+
+  @EventHandler
+  public void onPlayerDeath(MatchPlayerDeathEvent event) {
+    MatchPlayer victim = event.getVictim();
+    MatchPlayer murderer = null;
+    try {
+      murderer = event.getKiller().getParty().getPlayer(event.getKiller().getId());
+    } catch (NullPointerException ignored) {
+    }
+
+    playerStats victimStats = allPlayerStats.get(event.getVictim().getBukkit().getName());
+    if (victimStats == null) {
+      allPlayerStats.put(victim.getBukkit().getName(), new playerStats());
+      victimStats = allPlayerStats.get(event.getVictim().getBukkit().getName());
+    }
+    victimStats.onDeath();
+
+    playerStats finalVictimStats = victimStats;
+    // Without this delay the "Press LShift to dismount" message from the death minecart will
+    // override the stats message.
+    event
+        .getMatch()
+        .getScheduler(MatchScope.LOADED)
+        .runTaskLater(
+            0,
+            () -> {
+              NMSHacks.sendHotbarMessage(
+                  victim.getBukkit(), finalVictimStats.getBasicStatsMessage());
+            });
+
+    if (murderer != null
+        && PlayerRelation.get(victim.getParticipantState(), murderer.getState())
+            != PlayerRelation.ALLY
+        && PlayerRelation.get(victim.getParticipantState(), murderer) != PlayerRelation.SELF) {
+      playerStats murdererStats = allPlayerStats.get(murderer.getBukkit().getName());
+      if (murdererStats == null) {
+        allPlayerStats.put(murderer.getBukkit().getName(), new playerStats());
+        murdererStats = allPlayerStats.get(murderer.getBukkit().getName());
```

Comment:
I see your are using `murderer.getBukkit().getName()` and `event.getVictim().getBukkit().getName()` in this method. Can you make local variables in this method to make the code cleaner?